src/ui/tools/filterutils.js

```diff
@@ -19,3 +19,42 @@ export function filterIsPersisted (filter, persistedFilter) {
   }
   return isEqual(filter, persistedFilter);
 }
+
+/**
+ * Given a filter, return an array of all it's descendants, including itself, that
+ * filter on the given fieldId.
+ *
+ * @param {string} fieldId
+ * @param {Filter} persistedFilter
+ * @returns {Array<Filter>}
+ */
+export function findSimpleFiltersWithFieldId (fieldId, persistedFilter) {
+  const childFilters =
+    persistedFilter[FilterCombinators.AND] || persistedFilter[FilterCombinators.OR];
+  if (childFilters) {
+    return childFilters.flatMap(
+      childFilter => findSimpleFiltersWithFieldId(fieldId, Filter.from(childFilter)));
+  }
+  if (Filter.from(persistedFilter).getFilterKey() === fieldId) {
+    return [ persistedFilter ];
+  }
+  return [];
+}
+
+/**
+ * Finds a persisted range filter for the given fieldId, and returns its contents.
+ *
```

Comment:
nit: can we add jsdocs for the params? possible to have them in the same order as fn above?